opentelemetry-sdk/src/trace/span_processor.rs

```diff
@@ -632,14 +678,21 @@ impl SpanProcessor for BatchSpanProcessor {
     /// Shuts down the processor.
     fn shutdown_with_timeout(&self, timeout: Duration) -> OTelSdkResult {
         let dropped_spans = self.dropped_spans_count.load(Ordering::Relaxed);
-        let max_queue_size = self.max_queue_size;
         if dropped_spans > 0 {
+            // Log warning for observability (always happens)
             otel_warn!(
-                name: "BatchSpanProcessor.SpansDropped",
-                dropped_span_count = dropped_spans,
-                max_queue_size = max_queue_size,
-                message = "Spans were dropped due to a queue being full. The count represents the total count of spans dropped in the lifetime of this BatchSpanProcessor. Consider increasing the queue size and/or decrease delay between intervals."
+                name: "BatchSpanProcessor.Shutdown",
+                dropped_spans = dropped_spans,
+                max_queue_size = self.max_queue_size,
+                message = "Spans were dropped due to a full queue. The count represents the total count of span records dropped in the lifetime of the BatchSpanProcessor. Consider increasing the queue size and/or decrease delay between intervals."
             );
+
+            // Also return error so user code can handle it programmatically
```

Comment:
this feels incorrect. This is the `shutdown()` call, and that method is not failing here to warrant returning an `Err` - having items dropped due to buffer full is not a failure of `shutdown` itself.